lib/rules/require-sorted-object-keys.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Requires sorted object keys.
+ *
+ * Types: `Boolean` or `String`
+ *
+ * Values:
+ *  - `true` (alias to `asc`)
+ *  - `"asc"`: requires sorting in ascending order
+ *  - `"desc"`: requires sorting in descending order
+ *
+ * #### Example
+ *
+ * ```js
+ * "requireSortedObjectKeys": "asc"
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * var x = {
+ *  x: 'foo',
+ *  y: 'bar'
+ * }
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * var x = {
+ *  y: 'foo',
+ *  x: 'bar'
+ * }
+ * ```
+ */
+
+function asc(a, b) {
+    if (/[_]/.test(a)) {
+        return -1;
+    }
+
+    if (a < b) {
+        return -1;
+    }
+
+    if (a > b) {
+        return 1;
+    }
+
+    return 0;
+}
+
+function desc(a, b) {
```

Comment:
Since there are just two options (asc and desc) why not use `asc` in `desc`? Like `return -asc(a);`